public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx

```diff
@@ -226,55 +194,59 @@ export class ConfigEditor extends PureComponent<Props, State> {
     if (logAnalyticsSubscriptions && logAnalyticsSubscriptions.length > 0) {
       this.setState({ logAnalyticsSubscriptions });
 
-      this.state.config.jsonData.logAnalyticsSubscriptionId =
-        this.state.config.jsonData.logAnalyticsSubscriptionId || logAnalyticsSubscriptions[0].value;
+      this.props.options.jsonData.logAnalyticsSubscriptionId =
+        this.props.options.jsonData.logAnalyticsSubscriptionId || logAnalyticsSubscriptions[0].value;
     }
 
-    if (this.state.config.jsonData.logAnalyticsSubscriptionId) {
+    if (this.props.options.jsonData.logAnalyticsSubscriptionId) {
       await this.getWorkspaces();
     }
   };
 
   getWorkspaces = async () => {
-    const sameAs = this.state.config.jsonData.azureLogAnalyticsSameAs && this.state.config.jsonData.subscriptionId;
-    if (!sameAs && !this.state.config.jsonData.logAnalyticsSubscriptionId) {
+    const sameAs = this.props.options.jsonData.azureLogAnalyticsSameAs && this.props.options.jsonData.subscriptionId;
+    if (!sameAs && !this.props.options.jsonData.logAnalyticsSubscriptionId) {
       return;
     }
 
     const logAnalyticsWorkspaces = await this.loadWorkspaces(
-      sameAs ? this.state.config.jsonData.subscriptionId : this.state.config.jsonData.logAnalyticsSubscriptionId
+      sameAs ? this.props.options.jsonData.subscriptionId : this.props.options.jsonData.logAnalyticsSubscriptionId
     );
 
     if (logAnalyticsWorkspaces.length > 0) {
       this.setState({ logAnalyticsWorkspaces });
```

Comment:
We're setting state here as part of Promise chaining this could lead to trying to set state on an on loaded component. I think we have a [Cancelable Promise](https://github.com/grafana/grafana/blob/master/public/app/core/utils/CancelablePromise.ts) that you can use to prevent this from happening.